### PR TITLE
First shot at speaker invitation token

### PR DESCRIPTION
**Pull request by erdgeist**
This is a first shot at #358. Will do some translations and UI fixes.

### CODE REVIEW

**Comment by erdgeist** [comment]
This now in a state where I'm a little clueless what exactly I need to test for. I assume that the has_secure_token logic just works. Would that be an integration test or a controller test for adding just works with tokens?

**Comment by manno** [comment]
Since it's limited to one controller I think a controller test is a good start. If you want to test the whole procedure, probably integration is better suited. But whatever is easier is fine.

File: app/controllers/cfp/events_controller.rb [hunk 1]
```diff
@@ -83,6 +83,17 @@ def confirm
     end
   end
 
+  def join
+    @token = params[:token] || ''
+    @event = @token.blank? ? nil : Event.find_by(invite_token: @token)
+    if request.post?
+      return redirect_to cfp_join_event_path, flash: { error: t('cfp.join_token_unknown', token: @token) } if @event.nil?
```
Comment (artemasmith) [inline app/controllers/cfp/events_controller.rb]: It's quite complex logic. I think it's better to split it
```ruby
def join
 return unless request.post?
 @token = params[:token] || ''
 @event = @token.blank? ? nil : Event.find_by(invite_token: @token)
redirect_on_invalid_event
redirect_on_already_created
@event.event_people << EventPerson.new(person: current_user.person, event_role: 'speaker')
redirect_to edit_cfp_event_path(@event), notice: t('cfp.join_success')
end

private

def redirect_on_invalid_event
    return redirect_to cfp_join_event_path, flash: { error: t('cfp.join_token_unknown', token: @token) } if @event.nil?
 end

def redirect_on_already_created
     return redirect_to edit_cfp_event_path(@event), notice: t('cfp.join_token_already_used') if @event.people.exists?(current_user.person.id)
end

Comment (erdgeist) [reply app/controllers/cfp/events_controller.rb]: The returns from redirect_on_invalid_event and redirect_on_already_created don't propagate up to the join function, so this does not work.

File: app/controllers/cfp/events_controller.rb [hunk 2]
```diff
@@ -83,6 +83,17 @@ def confirm
     end
   end
 
+  def join
+    @token = params[:token] || ''
+    @event = @token.blank? ? nil : Event.find_by(invite_token: @token)
+    if request.post?
```
[the remark this answers is not in the document]
Comment (erdgeist) [reply app/controllers/cfp/events_controller.rb]: Yep.

File: app/controllers/cfp/events_controller.rb [hunk 3]
```diff
@@ -86,11 +86,16 @@ def confirm
...
     @token = params[:token] || ''
     @event = @token.blank? ? nil : Event.find_by(invite_token: @token)
-    if request.post?
-      return redirect_to cfp_join_event_path, flash: { error: t('cfp.join_token_unknown', token: @token) } if @event.nil?
-      return redirect_to edit_cfp_event_path(@event), notice: t('cfp.join_token_already_used') if @event.people.exists?(current_user.person.id)
-      @event.event_people << EventPerson.new(person: current_user.person, event_role: 'speaker')
-      return redirect_to edit_cfp_event_path(@event), notice: t('cfp.join_success')
+    return unless request.post?
+
+    if @event.nil?
```
Comment (artemasmith) [inline app/controllers/cfp/events_controller.rb]: Looks like more readable:)

Comment (erdgeist) [reply app/controllers/cfp/events_controller.rb]: Thank you. Important remark.

**Pull request closed without merging**